frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx

```diff
@@ -59,7 +59,7 @@ const ImageSearch: React.FC = () => {
             !values.application.name &&
               setFieldValue('application.name', `${getSuggestedName(name)}-app`);
             // set default port value
-            const targetPort = _.head(ports);
+            const targetPort = !values.route.targetPort && _.head(ports);
```

Comment:
Won't this conflict in the non-edit scenario where the user first searches and image and selects a target port, then changes the image only to have the old target port selected that may not be in the list of ports?

Reply:
So in this case what is happening target-port dropdown gets reset and shows `Select a target port` but in the form `targetPort` value is the same as old target-port so if the user doesnt change the value from the dropdown route will be created with the wrong port. I have a fix for this. Updating the PR.

Reply:
Isn't your fix still an issue in the edit form the next time someone changes the image?
Try changing from `openshift/hello-openshift` to `wordpress` and observe that no option gets selected in the dropdown.
I think you need a better way to distinguish between initial load and subsequent searches.

Reply:
Hmm true. Trying to find out a better solution for this issue.

Reply:
I did something similar to what I was doing for distinguishing between internal registry fields initial load and subsequent dropdown changes. Updated the PR.